### PR TITLE
Remove intermediate mavlink functions

### PR DESCRIPTION
**Pull request by peterbarker**
Pulling characters from the uart is something we do a lot of.  We should not unnecessarily make it slower.

### CODE REVIEW

**Comment by WickedShell** [comment]
Although I'd note this can be squashed to one commit :)

**Comment by peterbarker** [comment]
re: include problem... let me just quote @WickedShell for a moment...

```
Michael, 14:51
If you can remove the HAL extern from GCS_MAVLink do you still need to include hal.h?


14:51
shrug.  Probably :-)
Trying to remove all three includes...
```

@tridge This shows a problem I've seen before with `#define`s leaking from the ChibiOS build.  

Should I just put the `#include`s back in place and forget this ever happened?

**Comment by peterbarker** [comment]
```
[416/433] Compiling APMrover2/mode.cpp
In file included from ../../APMrover2/mode.h:4:0,
                 from ../../APMrover2/mode.cpp:1:
../../APMrover2/defines.h:8:18: error: expected identifier before numeric constant
 #define DISABLED 0
                  ^
../../APMrover2/defines.h:12:17: note: in expansion of macro 'DISABLED'
 #define DISABLE DISABLED
                 ^~~~~~~~
../../modules/ChibiOS/os/common/ext/CMSIS/ST/STM32F4xx/stm32f4xx.h:204:3: note: in expansion of macro 'DISABLE'
   DISABLE = 0U, 
   ^~~~~~~
compilation terminated due to -Wfatal-errors.

```

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -891,7 +891,7 @@ GCS_MAVLINK::update(uint32_t max_time_us)
     uint16_t nbytes = comm_get_available(chan);
```
Comment (OXINARF) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: `comm_get_available` is only used here so it's unneeded as well.

**Comment by peterbarker** [comment]
On Wed, 23 May 2018, Francisco Ferreira wrote:

> @OXINARF commented on this pull request.
> 
> Our include situation is horrible... we really need to review that part.

Yeah, but not as part of this PR.  I've fixed the Rover includes.

> > @@ -891,7 +891,7 @@ GCS_MAVLINK::update(uint32_t max_time_us)
>      uint16_t nbytes = comm_get_available(chan);
> 
> comm_get_available is only used here so it's unneeded as well.

Yeah, I thought about it.  But the locked mask is specific to 
GCS_MAVLink.cpp, and adding stuff to query/manipulate it seems a bad idea. 
I'd prefer to have these as separate bits within the GCS_MAVLINK instance 
- but the way the GPS library interacts makes it non-trivial.  Something 
for later.

**Pull request merged**